code/messaging/src/test/java/com/adobe/marketing/mobile/MessagingInternalHandleRulesResponseContentTests.java

```diff
@@ -0,0 +1,613 @@
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.services.ui.AEPMessage;
+import com.adobe.marketing.mobile.services.ui.FullscreenMessageDelegate;
+import com.adobe.marketing.mobile.services.ui.MessageSettings;
+import com.adobe.marketing.mobile.services.ui.UIService;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Event.class, MobileCore.class, ServiceProvider.class, ExtensionApi.class, ExtensionUnexpectedError.class, MessagingState.class, App.class, Context.class})
+public class MessagingInternalHandleRulesResponseContentTests {
+    private final Map<String, Object> mobileParameters = new HashMap<String, Object>() {
+        {
+            put(MessagingConstants.EventDataKeys.MobileParametersKeys.SCHEMA_VERSION, "version");
+            put(MessagingConstants.EventDataKeys.MobileParametersKeys.VERTICAL_ALIGN, "center");
+            put(MessagingConstants.EventDataKeys.MobileParametersKeys.HORIZONTAL_INSET, 0);
+            put(MessagingConstants.EventDataKeys.MobileParametersKeys.DISMISS_ANIMATION, "bottom");
+            put(MessagingConstants.EventDataKeys.MobileParametersKeys.UI_TAKEOVER, true);
+        }
+    };
+    private final Map<String, Object> mockConfigState = new HashMap<>();
+    private final Map<String, Object> mockIdentityState = new HashMap<>();
+    private final Map<String, Object> identityMap = new HashMap<>();
+    private final Map<String, Object> ecidMap = new HashMap<>();
+    private final List<Map> ids = new ArrayList<>();
+    private final byte[] base64EncodedBytes = "decisionScope".getBytes(StandardCharsets.UTF_8);
```

Comment:
`"decisionScope".getBytes(StandardCharsets.UTF_8);` will return the UTF-8 encoded bytes. Is there any reason to name this field as `base64EncodedBytes `?

Reply:
this is a mock return value for a call to EncodingService.base64Encode(). i've updated it to actually return the base64 encoded bytes rather than UTF8.